llumnix/global_scheduler/migration_scheduler.py

```diff
@@ -43,13 +59,38 @@ def __init__(self,
 
         self.num_instances = 0
         self.instance_id_set: Set[str] = set()
+        self.instance_type_id_set: Dict[InstanceType, Set[str]] = {instance_type: set() for instance_type in InstanceType}
+        self.constraint_prefill_instance_num = constraint_prefill_instance_num
         # instance info args
         self.instance_info: Dict[str, InstanceInfo] = None
-        self.sorted_instance_infos: List[InstanceInfo] = None
+        self.sorted_instance_infos: Dict[str, List[InstanceInfo]] = {instance_type: [] for instance_type in InstanceType}
 
-    def pair_migration(self) -> List[Tuple[str, str]]:
-        self._sort_instance_infos(descending=False)
-        return self.pair_migration_policy.pair_migration(self.sorted_instance_infos)
+    def pair_migration(self, pair_migration_type:str) -> List[Tuple[str, str]]:
+        sorted_src_instance_infos, sorted_dst_instance_infos = self._get_migration_instance_infos(pair_migration_type)
+        return self.pair_migration_policy.pair_migration(sorted_src_instance_infos, sorted_dst_instance_infos)
+
+    def _get_migration_instance_infos(self, pair_migration_type:str) -> Dict[str, InstanceInfo]:
```

Comment:
I suggest we simplify this function and abstract away common logic

Reply:
> We can also implement FilteringInstanceInfosPolicy here like PairMigrationPolicy to remove the if-else codes here and be more extensible. Besides, there is much duplicated codes here, if we can find someway to simplify the redcodes here, it would be better. And I know it's a bit difficult, it depends on you.



> I suggest we simplify this function and abstract away common logic

done

Reply:
lgtm